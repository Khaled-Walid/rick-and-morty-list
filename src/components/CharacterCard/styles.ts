import { SxProps } from '@mui/material';

export const CharacterCardStyles: Record<string, SxProps> = {
  card: {
    backgroundColor: 'rgb(60, 62, 68)',
    color: '#fff',
    width: { xs: '100%', sm: '580px' },
    display: 'flex',
    height: { xs: '480px', sm: '220px' },
    overflow: 'hidden',
    borderRadius: '0.5rem',
    flexDirection: { xs: 'column', sm: 'row' },
    alignItems: 'center',
  },
  charImg: {
    width: { xs: '280px', sm: '220px' },
    aspectRatio: '1/1',
    height: '100%',
  },
  cardContent: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-around',
    padding: '1rem',
    '&:last-child': {
      paddingBottom: '1rem',
    },
  },
  name: {
    fontWeight: '900',
    fontSize: '1.5rem',
    marginBottom: '-0.25rem',
    display: '-webkit-box',
    webkitLineClamp: '2',
    webkitBoxOrient: 'vertical',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
  },
  secText: {
    color: 'rgb(158, 158, 158)',
    lineHeight: '1rem',
  },
  statusDot: {
    display: 'inline-block',
    height: '0.5rem',
    width: '0.5rem',
    marginRight: '0.375rem',
    borderRadius: '50%',
  },
};
