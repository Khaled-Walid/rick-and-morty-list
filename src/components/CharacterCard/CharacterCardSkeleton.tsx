import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { Skeleton } from '@mui/material';
import { CharacterCardStyles as styles } from './styles';

export const CharacterCardSkeleton = (): JSX.Element => {
  return (
    <Card sx={styles.card}>
      <Skeleton variant="rectangular" sx={styles.charImg} />
      <CardContent sx={styles.cardContent}>
        <Typography variant="h4" sx={styles.name} style={{ display: 'block' }}>
          <Skeleton />
        </Typography>
        <Typography variant="subtitle1" gutterBottom>
          <Skeleton />
        </Typography>
        <Typography variant="body1" sx={styles.secText}>
          <Skeleton />
        </Typography>
        <Typography variant="body1" gutterBottom>
          <Skeleton />
        </Typography>
        <Typography variant="body1" sx={styles.secText}>
          <Skeleton />
        </Typography>
        <Typography variant="body1">
          <Skeleton />
        </Typography>
      </CardContent>
    </Card>
  );
};
