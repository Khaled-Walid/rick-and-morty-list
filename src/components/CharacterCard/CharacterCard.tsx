import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { CharacterCardSkeleton } from './CharacterCardSkeleton';
import { CharacterCardStyles } from './styles';
import { Box } from '@mui/material';

export interface CharacterCardData {
  image: string;
  name: string;
  status: 'Alive' | 'Dead' | 'unknown';
  species: string;
  gender: 'Female' | 'Male' | 'Genderless' | 'unknown';
  location: string;
  firstEpisode: string;
}

export interface CharacterCardProps {
  character: CharacterCardData | null;
}

const statusColors: Record<string, string> = {
  Alive: 'rgb(85, 204, 68)',
  Dead: 'rgb(214, 61, 46)',
  unknown: 'rgb(158, 158, 158)',
};

export const CharacterCard = ({
  character,
}: CharacterCardProps): JSX.Element => {
  if (character != null) {
    const { image, name, status, species, firstEpisode, gender, location } =
      character;
    const dotColor = statusColors[`${status}`];

    return (
      <Card sx={CharacterCardStyles.card}>
        <Box
          component="img"
          src={image}
          alt="Character"
          sx={CharacterCardStyles.charImg}
        />
        <CardContent sx={CharacterCardStyles.cardContent}>
          <Typography variant="h4" sx={CharacterCardStyles.name}>
            {name}
          </Typography>
          <Typography variant="subtitle1" gutterBottom>
            <Box
              sx={CharacterCardStyles.statusDot}
              style={{ backgroundColor: dotColor }}
            />
            {`${status} | ${species} | ${gender}`}
          </Typography>
          <Typography variant="body1" sx={CharacterCardStyles.secText}>
            Last known location:
          </Typography>
          <Typography variant="body1" gutterBottom>
            {location}
          </Typography>
          <Typography variant="body1" sx={CharacterCardStyles.secText}>
            First seen in:
          </Typography>
          <Typography variant="body1">{firstEpisode}</Typography>
        </CardContent>
      </Card>
    );
  } else {
    return <CharacterCardSkeleton />;
  }
};
