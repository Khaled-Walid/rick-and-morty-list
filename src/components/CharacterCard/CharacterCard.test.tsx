import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import matchMediaPolyfill from 'mq-polyfill';
import { adaptedCharactersMock } from '../../mocks/adaptedCharacters.mock';
import { CharacterCard } from './CharacterCard';

beforeAll(() => {
  matchMediaPolyfill(window);
  window.resizeTo = function resizeTo(width, height) {
    Object.assign(this, {
      innerWidth: width,
      innerHeight: height,
      outerWidth: width,
      outerHeight: height,
    }).dispatchEvent(new this.Event('resize'));
  };
});

beforeEach(() => {
  window.resizeTo(1920, 1080);
});

describe('CharacterCard', () => {
  it('renders data properly on big screens', () => {
    const rendered = render(
      <CharacterCard character={adaptedCharactersMock[0].character} />,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('renders data properly on small screens', () => {
    window.resizeTo(375, 667);
    const rendered = render(
      <CharacterCard character={adaptedCharactersMock[0].character} />,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('renders loading skeleton properly on big screens', () => {
    const rendered = render(<CharacterCard character={null} />);

    expect(rendered).toMatchSnapshot();
  });

  it('renders loading skeleton properly on small screens', () => {
    window.resizeTo(375, 667);
    const rendered = render(<CharacterCard character={null} />);
    expect(rendered).toMatchSnapshot();
  });
});
