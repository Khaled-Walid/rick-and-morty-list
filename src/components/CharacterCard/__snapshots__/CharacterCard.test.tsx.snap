// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard renders data properly on big screens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="Character"
          class="MuiBox-root css-b5dlu6"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <div
          class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
          >
            Rick Sanchez
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
          >
            <div
              class="MuiBox-root css-lxhyqj"
              style="background-color: rgb(85, 204, 68);"
            />
            Alive | Human | Male
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            Last known location:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            Citadel of Ricks
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            First seen in:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Pilot
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="Character"
        class="MuiBox-root css-b5dlu6"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
        >
          Rick Sanchez
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-lxhyqj"
            style="background-color: rgb(85, 204, 68);"
          />
          Alive | Human | Male
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          Last known location:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          Citadel of Ricks
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          First seen in:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Pilot
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterCard renders data properly on small screens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="Character"
          class="MuiBox-root css-b5dlu6"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <div
          class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
          >
            Rick Sanchez
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
          >
            <div
              class="MuiBox-root css-lxhyqj"
              style="background-color: rgb(85, 204, 68);"
            />
            Alive | Human | Male
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            Last known location:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            Citadel of Ricks
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            First seen in:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Pilot
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="Character"
        class="MuiBox-root css-b5dlu6"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
        >
          Rick Sanchez
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-lxhyqj"
            style="background-color: rgb(85, 204, 68);"
          />
          Alive | Human | Male
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          Last known location:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          Citadel of Ricks
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          First seen in:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Pilot
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterCard renders loading skeleton properly on big screens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-elit98-MuiSkeleton-root"
        />
        <div
          class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
            style="display: block;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-elit98-MuiSkeleton-root"
      />
      <div
        class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
          style="display: block;"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterCard renders loading skeleton properly on small screens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-elit98-MuiSkeleton-root"
        />
        <div
          class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
            style="display: block;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1295pbh-MuiPaper-root-MuiCard-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-elit98-MuiSkeleton-root"
      />
      <div
        class="MuiCardContent-root css-p5jh9q-MuiCardContent-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-cdjqzm-MuiTypography-root"
          style="display: block;"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8b1rag-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          />
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
